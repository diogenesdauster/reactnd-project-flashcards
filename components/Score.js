import React, { Component } from "react";
import {
  View,
  Text,
  StyleSheet,
  Animated,
  TouchableOpacity
} from "react-native";
import PropTypes from "prop-types";
import { clearLocalNotification, setLocalNotification } from "../utils/helpers";
import { white, lightblue2, lightblue3 } from "../utils/colors";

class Score extends Component {
  static propTypes = {
    score: PropTypes.string.isRequired,
    startAgain: PropTypes.func.isRequired,
    goBackToDeck: PropTypes.func.isRequired
  };

  state = {
    scoreAnimeted: new Animated.Value(0)
  };

  componentDidMount() {
    clearLocalNotification().then(setLocalNotification);
    this.state.scoreAnimeted.addListener(({ value }) => {
      const score = Number(value).toFixed(2);
      this.setState({ score });
    });
    Animated.timing(this.state.scoreAnimeted, {
      toValue: this.props.score,
      duration: 500,
      delay: 1000
    }).start();
  }
  componentWillUnmount() {
    this.state.scoreAnimeted.removeAllListeners();
  }
  render() {
    const { score, startAgain, goBackToDeck } = this.props;
    return (
      <View style={styles.container}>
        <Text style={{ fontSize: 32 }}>Your score is:</Text>
        <Animated.Text style={{ fontSize: 40, marginBottom: 20 }}>
          {score || "0.00"}%
        </Animated.Text>
        <TouchableOpacity
          style={[styles.button, { backgroundColor: lightblue2 }]}
          onPress={startAgain}
        >
          <Text style={{ fontSize: 32, color: white }}>Start Again</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.button, { backgroundColor: lightblue3 }]}
          onPress={goBackToDeck}
        >
          <Text style={{ fontSize: 32, color: white }}>Back to Deck</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default Score;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  button: {
    width: 250,
    height: 60,
    marginBottom: 10,
    justifyContent: "center",
    alignItems: "center",
    shadowRadius: 3,
    shadowOpacity: 0.8,
    shadowColor: "rgba(0, 0, 0, 0.24)",
    shadowOffset: {
      width: 0,
      height: 3
    }
  }
});
