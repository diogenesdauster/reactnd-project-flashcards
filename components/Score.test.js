import "react-native";
import React from "react";
import Score from "./Score";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

// static propTypes = {
//     score: PropTypes.string.isRequired,
//     startAgain: PropTypes.func.isRequired,
//     goBackToDeck: PropTypes.func.isRequired
//   };

it("renders Score correctly", () => {
  const startAgain = jest.fn();
  const goBackToDeck = jest.fn();

  const tree = renderer
    .create(
      <Score
        score={"100"}
        startAgain={startAgain}
        goBackToDeck={goBackToDeck}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
