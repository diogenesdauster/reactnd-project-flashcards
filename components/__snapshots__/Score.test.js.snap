// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Score correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    Your score is:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 40,
        "marginBottom": 20,
      }
    }
  >
    0.00
    %
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3F51B5",
        "height": 60,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "shadowColor": "rgba(0, 0, 0, 0.24)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 3,
        "width": 250,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 32,
        }
      }
    >
      Start Again
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00BCD4",
        "height": 60,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "shadowColor": "rgba(0, 0, 0, 0.24)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 3,
        "width": 250,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 32,
        }
      }
    >
      Back to Deck
    </Text>
  </View>
</View>
`;
